### DISCUSSION TITLE
How to optimally start *tmux* in  every Termux session?

### DISCUSSION THREAD

**Discussion by ghost** [termux/termux-app#4123] [Q&A] [answered]
I've tmux installed and want to launch it with every Termux session. [A suggestion shared by the /r/termux moderator](https://unix.stackexchange.com/questions/43601/how-can-i-set-my-default-shell-to-start-up-tmux) doesn't work (I've restarted; I don't see other relevant info).

>Add to `~/.zlogin` end:
>
>`if [ -z "$TMUX" ]; then`
>`exec tmux`
>`fi`
>
>Checking for `$TMUX` variable is done to ensure that no nested tmux instance will be started on login shells. `exec` will replace current process with a new one, i.e. current shell will be terminated and tmux instance with shell inside will be started, so you won't need to exit twice.

---

see for Linux in general: https://unix.stackexchange.com/questions/43601/how-can-i-set-my-default-shell-to-start-up-tmux

**Comment by sylirre** [comment, marked as answer]
`~/.zlogin` specific to zsh shell but default shell of Termux is bash. It uses `~/.bash_profile`, in this file you can add:
```
# Don't run code below if shell is non-interactive
[[ $- != *i* ]] && return

# Start tmux if not running already in current shell session.
if [ -z "$TMUX" ]; then
    exec tmux
fi
```

If you want all Termux sessions to share single tmux instance use this instead:
```
# Don't run code below if shell is non-interactive
[[ $- != *i* ]] && return

# If not already under tmux, try attaching to existing instance. If no tmux instance exist, start new.
if [ -z "$TMUX" ]; then
    tmux attach 2>/dev/null || tmux
    exit
fi
```
*Side effect of using single tmux instance: ctrl-d will terminate all Termux sessions if tmux had only one.*

**Comment by ghost** [comment]
Sorry for confusing the shells. The top answer on [unix.stackexchange.com](https://unix.stackexchange.com/questions/43601/how-can-i-set-my-default-shell-to-start-up-tmux) is also functional (restart Termux after saving).
